commons-numbers-gamma/src/main/java/org/apache/commons/numbers/gamma/BoostErf.java

```diff
@@ -0,0 +1,674 @@
+        //
+        // Begin by testing for domain errors, and other special cases:
+        //
+        if (z < 0 || z > 2 || Double.isNaN(z)) {
+            // Argument outside range [0,2] in inverse erfc function
+            return Double.NaN;
+        }
+        // Domain bounds must be detected as the implementation computes NaN.
+        // (log(q=0) creates infinity and the rational number is
+        // infinity / infinity)
+        if (z == (int) z) {
+            // z   return
+            // 2   -inf
+            // 1   0
+            // 0   inf
+            return z == 1 ? 0 : (1 - z) * Double.POSITIVE_INFINITY;
+        }
+
+        //
+        // Normalise the input, so it's in the range [0,1], we will
+        // negate the result if z is outside that range. This is a simple
+        // application of the erfc reflection formula: erfc(-z) = 2 - erfc(z)
+        //
+        double p;
+        double q;
+        double s;
+        if (z > 1) {
+            q = 2 - z;
+            p = 1 - q;
+            s = -1;
+        } else {
+            p = 1 - z;
+            q = z;
+            s = 1;
+        }
+
+        //
+        // And get the result, negating where required:
+        //
+        return s * erfInvImp(p, q);
+    }
+
+    /**
+     * Returns the inverse error function.
+     *
+     * @param z Value (in {@code [-1, 1]}).
+     * @return t such that {@code z = erf(t)}
+     */
+    static double erfInv(double z) {
+        //
+        // Begin by testing for domain errors, and other special cases:
+        //
+        if (z < -1 || z > 1 || Double.isNaN(z)) {
+            // Argument outside range [-1, 1] in inverse erf function
+            return Double.NaN;
+        }
+        // Domain bounds must be detected as the implementation computes NaN.
+        // (log(q=0) creates infinity and the rational number is
+        // infinity / infinity)
+        if (z == (int) z) {
+            // z   return
+            // -1  -inf
+            // -0  -0
+            // 0   0
+            // 1   inf
+            return z == 0 ? z : z * Double.POSITIVE_INFINITY;
+        }
+
+        //
+        // Normalise the input, so it's in the range [0,1], we will
+        // negate the result if z is outside that range. This is a simple
+        // application of the erf reflection formula: erf(-z) = -erf(z)
+        //
+        double p;
+        double q;
+        double s;
+        if (z < 0) {
+            p = -z;
+            q = 1 - p;
+            s = -1;
+        } else {
+            p = z;
+            q = 1 - z;
+            s = 1;
+        }
+        //
+        // And get the result, negating where required:
+        //
+        return s * erfInvImp(p, q);
+    }
+
+    /**
+     * Common implementation for inverse erf and erfc functions.
+     *
+     * @param p P-value
+     * @param q Q-value (1-p)
+     * @return the inverse
+     */
+    private static double erfInvImp(double p, double q) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 136 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.